scripts/language_model/word_language_model.py

```diff
@@ -354,30 +357,51 @@ def train():
             data_list = gluon.utils.split_and_load(data, context, batch_axis=1, even_split=True)
             target_list = gluon.utils.split_and_load(target, context, batch_axis=1, even_split=True)
             hiddens = detach(hiddens)
+            model.collect_params().zero_grad()
             Ls = []
             L = 0
             with autograd.record():
                 for j, (X, y, h) in enumerate(zip(data_list, target_list, hiddens)):
-                    output, h, encoder_hs, dropped_encoder_hs = model.forward(X, h)
+                    output, h, encoder_hs, dropped_encoder_hs = model(X, h)
                     l = joint_loss(output, y, encoder_hs, dropped_encoder_hs)
                     L = L + l.as_in_context(context[0]) / X.size
-                    Ls.append(l/X.size)
+                    Ls.append(l / X.size)
                     hiddens[j] = h
             L.backward()
```

Comment:
You can simply perform backward by doing for loop in Ls. No need to have Ls and L at the same time